test/lib/commands/link.js

```diff
@@ -514,3 +514,39 @@ t.test('--global option', async t => {
     'should throw an useful error'
   )
 })
+
+t.test('hash character in working directory path', async t => {
+  const testdir = t.testdir({
+    'global-prefix': {
+      lib: {
+        node_modules: {
+          a: {
+            'package.json': JSON.stringify({
+              name: 'a',
+              version: '1.0.0',
+            }),
+          },
+        },
+      },
+    },
+    'i_like_#_in_my_paths': {
+      'test-pkg-link': {
+        'package.json': JSON.stringify({
+          name: 'test-pkg-link',
+          version: '1.0.0',
+        }),
+      },
+    },
+  })
+  npm.globalDir = resolve(testdir, 'global-prefix', 'lib', 'node_modules')
+  npm.prefix = resolve(testdir, 'i_like_#_in_my_paths', 'test-pkg-link')
+
+  link.workspacePaths = null
```

Comment:
Previous test(s) were leaving workspaces setup without proper tear down. At the moment it was easiest way to fix that, as there is no some `link.clearWorkspaces` or `unsetWorkspaces`. Is it even legal to use that here?

Reply:
It's fine for now, this test isn't up to date on the newest mock npm object, and we can readdress this hack when we do update it.

Reply:
And by readdress I mean remove it since each test will be using its own individual real npm instance ;)